src/Controller/HomeController.php

```diff
@@ -0,0 +1,299 @@
+   *
+   * @Route("/forgotpassword", name = "forgotPassword")
+   *  THis route to take the user to the forgot password page.
+   *
+   *   @return Response
+   *     Returns Response to the forgot password page.
+   */
+  public function forgotPassword(Request $rq): Response
+  {
+    return $this->render("forgotpassword.html.twig");
+  }
+
+  /**
+   * This method is used for verify the user id entered by the user.
+   *
+   * @Route("/forgotpassword/checkuserid")
+   *  This route used to send message to the ajax function if user id
+   *  is valid or not.
+   *
+   *   @param Request $rq
+   *     Accepts the Request from the ajax call.
+   *
+   *   @return JsonResponse
+   *     Returns Json data to the calling ajax function.
+   */
+  public function verifyUser(Request $rq): JsonResponse
+  {
+   $message = $this->forgotPassword->checkUser($rq->request->get("userId"));
+   return new JsonResponse(json_encode([
+    "isValidUser" => $message,
+   ]));
+  }
+
+  /**
+   * This method is to send the otp to the user email id and returns
+   * the success message to the user.
+   *
+   * @Route("/forgotpassword/sendotp")
+   *  This route used to send message to the ajax function whether the mail sent
+   *  successfully or not.
+   *
+   *   @return JsonResponse
+   *     Returns Json data to the calling ajax function.
+   */
+  public function sendOtp(): JsonResponse
+  {
+    $message = $this->forgotPassword->sendOtp();
+    return new JsonResponse(json_encode([
+      "otpSendMessage" => $message,
+    ]));
+  }
+
+  /**
+   * This method verify the otp entered by the user and returns the
+   * message accordingly.
+   *
+   * @Route("/forgotpassword/verifyotp")
+   *  This route used to send message to the ajax function whether the
+   *  otp is correct or not.
+   *
+   *   @param Request $rq
+   *     Accepts the Request from the ajax call.
+   *
+   *   @return JsonResponse
+   *     Returns Json data to the calling ajax function.
+   */
+  public function verifyOtp(Request $rq): JsonResponse
+  {
+    $message = $this->forgotPassword->verifyOtp($rq->request->get('otp'));
+    return new JsonResponse(json_encode([
+      "verifyOtp" => $message,
+    ]));
+  }
+
+  /**
+   * This method is to reset the password of the user basically it stores
+   * the new password entered by the user in database of that perticular user
+   * and send a success message.
+   *
+   * @Route("/forgotpassword/reset")
+   *  This route used to reset the password send message to the ajax
+   *  function that the password reset successfully.
+   *
+   *   @param Request $rq
+   *     Accepts the Request from the ajax call.
+   *
+   *   @return JsonResponse
+   *     Returns Json data to the calling ajax function.
+   */
+  public function resetPassword(Request $rq): JsonResponse
+  {
+    $user = $this->userRepo->findOneBy([
+      "userId" => $rq->request->get("userid"),
+    ]);
+    $user->setPassword(md5($rq->request->get("password")));
```

Comment:
Dont you check for strong password validation?

Reply:
It is checked in frontend using jquery.

Reply:
You should do validations in the backend also.